src/main.cpp

```diff
@@ -84,7 +84,42 @@ void open_output_file(const std::string& a, const std::string& b) {
 
 /**************************************************************************************************/
 
-void process_orc_config_file(const char* bin_path_string) {
+template <typename T>
+T parse_enval(std::string&&) = delete;
+
+template <>
+std::string parse_enval(std::string&& x) {
+    assert(!x.empty());
+    return x;
+}
+
+template <>
+bool parse_enval(std::string&& x) {
+    assert(!x.empty());
+    return x != "0" && toupper(std::move(x)) != "FALSE";
+}
+
+template <>
+std::size_t parse_enval(std::string&& x) {
+    assert(!x.empty());
+    return std::max<int>(std::atoi(x.c_str()), 0);
+}
+
+template <typename T>
+T derive_configuration(const char* key,
+                       const toml::parse_result& settings,
+                       T&& fallback) {
+    T result = settings[key].value_or(fallback);
+    std::string envar = toupper(std::string("ORC_") + key);
+    if (const char* enval = std::getenv(envar.c_str())) {
+        result = parse_enval<T>(enval);
+    }
+    return result;
+}
```

Comment:
This routine is the heart of the PR. Instead of taking values from the settings file directly, `derive_configuration` will also look at the associated environment variable and use that if found.